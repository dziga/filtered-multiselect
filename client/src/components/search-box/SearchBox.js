import React, { Fragment } from 'react';
import './SearchBox.scss';

function SearchBox (props) {
  const {onSearch} = props 
  const onInput = (e) => onSearch(e.target.value)
  return (
    <Fragment>
      <input className="search" type="text" placeholder="Zoek op..." onChange={onInput} />
    </Fragment>
  );
}

export default SearchBox;
