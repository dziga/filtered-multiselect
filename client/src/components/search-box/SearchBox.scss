@import '../common/variables';

.search {
  border: 2px solid $border;
  border-radius: 5px;
  margin-bottom: 20px;
  padding: 10px;
  font-size: $font-medium;
  background: no-repeat 95% 50% url('./search-icon.svg');
  background-size: 15px 15px;
  background-color: white;
}