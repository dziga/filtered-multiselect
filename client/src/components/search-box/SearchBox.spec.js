import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SearchBox from './SearchBox';

test('shearch box should render', () => {
  const component = renderer.create(
    <SearchBox />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('search box should return search term', () => {
  const onSearch = jest.fn()
  const component = shallow((<SearchBox onSearch={onSearch}/>))
  const search = 'book';

  component.find('input').simulate('change', { target: { value: search } })
  expect(onSearch.mock.calls[0][0]).toEqual(search);
})
