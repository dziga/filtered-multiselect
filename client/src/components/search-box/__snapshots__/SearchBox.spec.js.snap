// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shearch box should render 1`] = `
<input
  className="search"
  onChange={[Function]}
  placeholder="Zoek op..."
  type="text"
/>
`;
